geopandas/geodataframe.py

```diff
@@ -194,6 +194,23 @@ def __init__(self, data=None, *args, geometry=None, crs=None, **kwargs):
             ):
                 raise ValueError(crs_mismatch_error)
 
+            if hasattr(geometry, "name") and geometry.name not in ("geometry", None):
+                # __init__ always creates geometry col named "geometry"
+                # but `set_geometry` respects the given series name
+                msg = (
+                    f"You have passed a Series named {geometry.name!r} as the active "
+                    f"geometry column to use, but the name {geometry.name!r} will be "
+                    "ignored as the GeoDataFrame constructor "
+                    'always creates a geometry column named "geometry". To silence '
+                    "this warning you can either call "
+                    'GeoDataFrame(..., geometry=ser.rename("geometry")) to keep the '
```

Comment:
wouldn't passing the underlying array be preferable to rename?

Reply:
I theory you could do either `.array` or `.rename("geometry")`, and those _can_ do something different in case the series has a different index than the data (in which case passing a GeoSeries does alignment, but passing `.array` will no longer do that alignment). 

That's a nuanced difference that will get too detailed to explain in a warning though ..

Reply:
Ah, true. Fine with the current suggestion then.

Reply:
That said, I think I might prefer suggesting `.array` if we keep a warning, because it is a bit shorter. And it feels strange to ask people to manually rename the series to "geometry" just because the result will use that name ("if the result will have that name anyway, then why do I have to rename it manually?")